runtime/parachains/src/disputes/slashing.rs

```diff
@@ -0,0 +1,905 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Dispute slashing pallet.
+//!
+//! The implementation relies on the `offences` pallet and
```

Comment:
A more detailed description would be really helpful.

Reply:
Let me know if [c22a943](https://github.com/paritytech/polkadot/pull/5535/commits/c22a94301b0ef7966efb29aa06ff7011da6be030) is good enough :)